cmd/logs.go

```diff
@@ -371,3 +461,158 @@ func logsDNS() {
+	var sb strings.Builder
+	sb.WriteString("\"query\":{\"bool\":{\"filter\":[")
+	for _, term := range f.term {
+		sb.WriteString("{")
+		sb.WriteString(term.toJSON())
+		sb.WriteString("},")
+	}
+	sb.WriteString("{")
+	sb.WriteString(f.rng.toJSON())
+	sb.WriteString("}]}}")
+	return sb.String()
+}
+
+type requestQuery struct {
+	BoolQuery boolQuery
+	From      int64
+	Size      int64
+	Source    []string
+}
+
+func (q *requestQuery) toJSON() string {
+	var sb strings.Builder
+	sb.WriteString("{")
+	sb.WriteString(q.BoolQuery.toJSON() + ",")
+	sb.WriteString(fmt.Sprintf("\"from\":%d,", q.From))
+	sb.WriteString(fmt.Sprintf("\"size\":%d,", q.Size))
+	sb.WriteString("\"sort\": {\"@timestamp\": \"desc\"},")
+	sb.WriteString("\"_source\":[")
+	for index, source := range q.Source {
+		if index != 0 {
+			sb.WriteString(", " + fmt.Sprintf("\"%s\"", source))
+		} else {
+			sb.WriteString(fmt.Sprintf("\"%s\"", source))
+		}
+	}
+	sb.WriteString("]}")
+	return sb.String()
+}
+
+func newRequestQuery(from, size int64, r rng, sources []string, terms []term) *requestQuery {
+	q := new(requestQuery)
+	q.BoolQuery.rng = r
+	q.BoolQuery.term = append(q.BoolQuery.term, terms...)
+	q.From = from
+	q.Size = size
+	q.Source = append(q.Source, sources...)
+	return q
+}
+
+func createElasticQueries(podName, containerName string) []requestQuery {
+	podTokens := strings.Split(podName, "-")
+	containerTokens := strings.Split(containerName, "-")
+	terms := setTerms(podTokens, containerTokens)
+	sources := []string{"@timestamp", "severity", "source", "log"}
+	rng := rng{TimeField: "@timestamp", Gte: int64(flags.sinceSeconds.Seconds())}
+	if rng.Gte == 0 {
+		rng.Gte = fourteenDaysInSeconds
+	}
+
+	numberOfRequests := int((flags.tail-1)/maxEsLogsPerQuery + 1)
+	queries := buildQueries(numberOfRequests, rng, sources, terms)
+	return queries
+}
+
+func setTerms(podTokens, containerTokens []string) []term {
+	terms := []term{}
+	for _, token := range podTokens {
+		terms = append(terms, term{Key: "kubernetes.pod_name", Value: token})
+	}
+	for _, token := range containerTokens {
+		if token != "" {
+			terms = append(terms, term{Key: "kubernetes.container_name", Value: token})
+		}
+	}
+	return terms
+}
+
+func buildQueries(numberOfRequests int, r rng, sources []string, terms []term) []requestQuery {
+	queries := []requestQuery{}
+	from := int64(0)
+	if flags.tail == -1 {
+		flags.tail = maxEsLogsPerQuery
+	}
+	for i := 0; i < numberOfRequests; i++ {
+		size := flags.tail - from
+		if size > maxEsLogsPerQuery {
+			size = maxEsLogsPerQuery
+		}
+		queries = append(queries, *newRequestQuery(from, size, r, sources, terms))
+		from += (maxEsLogsPerQuery + 1)
+	}
+	return queries
+}
+
+func (response logResponse) String() string {
+	//var output strings.Builder
+	output := make([]string, 0)
+	for _, hit := range response.Hits.Hits {
+		source := hit.Source
+		output = append(output, fmt.Sprintf("%v %v %v | %v", source.Timestamp, source.Severity, source.Source, source.Log))
```

Comment:
If you want the output to be aligned, you could check https://golang.org/pkg/text/tabwriter/